keps/sig-apps/20190318-PodDisruptionBudget-graduation-to-stable.md

```diff
@@ -93,6 +93,18 @@ but will still require `DisruptionsAllowed` to be larger than 0 for it to be evi
 arise from this. For example if we have a PDB with `MinAvailable  = 1` and 10 pods that are all in the CrashLoop state 
 (`Running`, but not `Ready`), we will not be allowed to evict any of the pods.
 
+To fix this, we will unify the logic between the disruption controller and the eviction REST handler so that a pod that 
+is not considered ready by the controller will always be allowed to be evicted (even if DisruptionsAllowed is 0). 
```

Comment:
We should still respect e.g. `maxUnavailable`, right? If there is 3 database pods with `maxUnavailable==1` and 2 of them blip on readiness, you should be able to delete just 1, not 2.

Reply:
So with this proposal, both pods that are will be eligible for eviction without looking at the PDB if the `Ready` condition is not `True`. For a pod that is crashlooping, it will be `Ready` intermittently for short periods of time. As time goes by, the backoff for restarting pods means that they will not be ready most of the time. In the situation you are describing, they will most likely both be evicted. 
This comes back to the core question here. If a pod is not ready (which means among other things that it is not targeted by services), are there situations where it is better to keep it around in the crashlooping state vs removing it and rely on the controller to bring up a new pod to replace it. 
I think it would be safe to remove it since the user/admin wants the pod removed from the node, and by using conditional deletes, we will make sure that the pod is in fact not in the ready state by the time it is deleted. And a controller will bring up a new pod to replace it.

Reply:
We still have to ensure we have enough healthy replicas if a particular pod is Ready==False.  Pods can transition to ready/not ready for a number of reasons.  Ensuring we instead have enough ready pods will potentially prevent data loss.

My PR addresses this issue: https://github.com/kubernetes/kubernetes/pull/94381